### PR TITLE
CI-CD Updates

### PR DESCRIPTION
**Pull request by Skptak**
Description
-----------
CI-CD Updates to use new versions of the CI-CD-Github-Actions
Individual PRs can be seen at:
[Complexity Check](https://github.com/FreeRTOS/CI-CD-Github-Actions/pull/70)
[Coverage-Cop](https://github.com/FreeRTOS/CI-CD-Github-Actions/pull/71)
[Executable Monitor](https://github.com/FreeRTOS/CI-CD-Github-Actions/pull/72)
[CSpell](https://github.com/FreeRTOS/CI-CD-Github-Actions/pull/75)
[Uncrustify Formatting Update](https://github.com/FreeRTOS/CI-CD-Github-Actions/pull/76)
[Link Verifier](https://github.com/FreeRTOS/CI-CD-Github-Actions/pull/81)

Test Steps
-----------

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [x] I have modified and/or added unit-tests to cover the code changes in this Pull Request.

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: cspell.config.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+---
+$schema: https://raw.githubusercontent.com/streetsidesoftware/cspell/main/cspell.schema.json
+version: '0.2'
+# Allows things like stringLength
+allowCompoundWords: true
```
Comment (bradleysmith23) [inline cspell.config.yaml:5]: Are both uses of allowCompoundWords necessary?

File: cspell.config.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+useGitignore: true
+languageSettings:
+  - allowCompoundWords: true
+    caseSensitive: false
+    enabled: true
+    languageId: c
+    locale: "*"
+# Could split this up? And do a dictionary for each repo?
```
Comment (bradleysmith23) [inline cspell.config.yaml:13]: Remove unneeded comment

**Review by bradleysmith23 — CHANGES_REQUESTED** [review]

**Review by tony-josi-aws — DISMISSED** [review]

File: MISRA.md [hunk 3]
```diff
@@ -1,190 +1,204 @@
...
-       unaligned access. But, in case of FreeRTOS+TCP, the buffer in which the
-       packets are stored are always aligned to a 4 byte word boundary with an
-       offset of 2 bytes. The reason for this 2 byte offset is that the
-       ethernet header is of 14 (12 + 2) bytes. Thus, everything except the
-       ethernet header is properly aligned. There is one alignment exception,
-       which is the sender protocol address in the ARP Header. To combat that,
-       the sender protocol address field is declared as an array of 4 bytes
-       instead of a `uint32_t`.
-       More details can be found [here](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/pull/512#pullrequestreview-1035211706).
+- MISRA C-2012 Rule 11.3 The data received/sent by the IP stack is represent as
```
[the remark this answers is not in the document]
Comment (Skptak) [reply MISRA.md]: Fixed

**Comment by Skptak** [comment]
All PRs into FreeRTOS-Plus-TCP are going to fail until https://github.com/FreeRTOS/FreeRTOS-Kernel/pull/778 is merged in.

**Review by tony-josi-aws — DISMISSED** [review]

File: README.md [hunk 4]
```diff
@@ -36,8 +36,8 @@ For running the script, you will need Python version > 3.7. You can download/ins
 
 Once python is downloaded and installed, you can verify the version from your terminal/command window by typing `python --version`.
 
-To run the script, you should switch to the FreeRTOS-Plus-TCP directory that was created using the [Cloning this repository](#cloning-this-repository) step above.
```
Comment (bradleysmith23) [inline README.md:39]: Should this have been removed?

Comment (Skptak) [reply README.md:39]: Yeah, those instructions no longer exist in this README

Comment (bradleysmith23) [inline README.md:39]: Should this have been deleted?

**Review by archigup — DISMISSED** [review]

**Review by bradleysmith23 — APPROVED** [review]
Left a comment on one chang

**Review by bradleysmith23 — APPROVED** [review]

**Review by archigup — APPROVED** [review]

**Pull request merged**